reconcile/aus/ocm_upgrade_scheduler.py

```diff
@@ -94,3 +109,31 @@ def get_ocm_env_upgrade_specs(
             org_name: OrganizationUpgradeSpec(org=specs[0].ocm, specs=specs)
             for org_name, specs in specs_per_org.items()
         }
+
+    def expose_remaining_soak_day_metrics(
+        self,
+        ocm_env: str,
+        upgrade_policies: list[ConfiguredUpgradePolicy],
+        version_data: VersionData,
+        available_upgrades: dict[str, list[str]],
+    ) -> None:
+        for up in upgrade_policies:
+            upgrades = available_upgrades.get(up.cluster) or []
+            if not upgrades:
+                continue
+
+            workload_soaking_upgrades = [
+                aus.soaking_days(version_data, upgrades, wl, False)
+                for wl in up.workloads
+            ]
+            for version in upgrades or []:
+                soaks = [s.get(version, 0) for s in workload_soaking_upgrades]
+                metrics.set_gauge(
+                    AUSClusterVersionRemainingSoakDaysMetric(
```

Comment:
Should we use a scoped transactional metric context here in order to automatically remove policies/clusters that disappeared?

Reply:
also `version` would disappear

Reply:
all exposed metrics are under a scoped transaction

https://github.com/app-sre/qontract-reconcile/blob/bf6f6e8053ad14a439029a116012a53101823104/reconcile/aus/base.py#L78

Reply:
ah did not go up the stack enough! all good!